url24d793285d_auzll_gengweibo_tgz-pJ8-org_json_TestJ8/src/com/gengweibo/weibo/WeiboQQ.java

```diff
@@ -0,0 +1,222 @@
+/**
+ * http://auzll.iteye.com/
+ */
+package com.gengweibo.weibo;
+
+import static com.gengweibo.weibo.Weibo.Method.GET;
+import static com.gengweibo.weibo.Weibo.Method.POST;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+import net.oauth.OAuth;
+import net.oauth.OAuth.Parameter;
+import net.oauth.OAuthConsumer;
+import net.oauth.OAuthMessage;
+import net.oauth.ParameterStyle;
+import net.oauth.client.OAuthClient;
+import net.oauth.client.URLConnectionClient;
+import net.oauth.http.HttpMessage;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import com.google.inject.internal.Lists;
+
+/**
+ * @author auzll@msn.com
+ * @since 2011-3-16
+ */
+public class WeiboQQ extends Weibo implements IWeibo {
+    /**  */
+    private static final long serialVersionUID = 5345772573791707652L;
+
+    public WeiboQQ() {
+        super("https://open.t.qq.com/cgi-bin/request_token",
+                "https://open.t.qq.com/cgi-bin/authorize",
+                "https://open.t.qq.com/cgi-bin/access_token",
+                "http://open.t.qq.com/api/", OAuthProperty.getCallbackUrl(),
+                OAuthProperty.CONSUMER_KEY_QQ,
+                OAuthProperty.CONSUMER_SECRET_QQ, WeiboType.T_QQ);
+    }
+
+    public Response favoritesCreate(IParam param) {
+        String id = param.getParamValue("statusId");
+        return sendRequest(toRequestParam("format", "json").add("id", id),
+                urlResource + "fav/addt", POST);
+    }
+
+    public Response friendshipsCreate(IParam param) {
+        String name = param.getParamValue("userId");
+        return sendRequest(toRequestParam("format", "json").add("name", name),
+                urlResource + "friends/add", POST);
+    }
+
+    public Response friendshipsDestroy(IParam param) {
+        String name = param.getParamValue("userId");
+        return sendRequest(toRequestParam("format", "json").add("name", name),
+                urlResource + "friends/del", POST);
+    }
+
+    public Response homeTimeline(IParam param) {
+        RequestParam reqParam = toRequestParam("reqnum",
+                parseLegalPageCount(param.getParamValue("count"))).add(
+                "format", "json");
+
+        if (null != param.getParamValue("pagetime")) {
+            reqParam.add("pagetime", param.getParamValue("pagetime"));
+        }
+
+        if (null != param.getParamValue("pageflag")) {
+            reqParam.add("pageflag", param.getParamValue("pageflag"));
+        }
+
+        /*
+         * pageflag 分页标识（0：第一页，1：向下翻页，2向上翻页） pagetime 本页起始时间（第一页
+         * 0，继续：根据返回记录时间决定）
+         */
+        return sendRequest(reqParam, urlResource + "statuses/home_timeline",
+                GET);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 63:80, 202:215.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.